File: backend/models/complaint.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Complaint struct {
	ID          uint           `gorm:"primaryKey" json:"id"`
	TenantID    uint           `json:"tenant_id"`
	Tenant      Tenant         `gorm:"foreignKey:TenantID" json:"tenant"`
	Title       string         `gorm:"size:100" json:"title"`
	Description string         `gorm:"type:text" json:"description"`
	Status      string         `gorm:"size:20" json:"status"` // pending, in_progress, resolved
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

File: rumatenant/lib/models/complaint.dart
class Complaint {
  final int id;
  final int tenantId;
  final String title;
  final String description;
  final String photoUrl;
  final String status;
  final DateTime createdAt;

  Complaint({
    required this.id,
    this.tenantId = 0,
    required this.title,
    required this.description,
    this.photoUrl = '',
    this.status = 'pending',
    DateTime? createdAt,
  }) : createdAt = createdAt ?? DateTime.now();

  factory Complaint.fromJson(Map<String, dynamic> json) {
    return Complaint(
      id: json['id'] ?? 0,
      tenantId: json['tenant_id'] ?? 0,
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      photoUrl: json['photo_url'] ?? '',
      status: json['status'] ?? 'pending',
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at']) ?? DateTime.now()
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {'title': title, 'description': description, 'photo_url': photoUrl};
  }
}
